device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -238,11 +369,22 @@ public void registerDeviceClient(DeviceClient deviceClient) throws InterruptedEx
      * <p>
      * @throws InterruptedException If the thread gets interrupted while waiting for the registration to succeed. This
      * will never be thrown if the multiplexing client is not open yet.
+     * @throws MultiplexingClientDeviceRegistrationFailedException If one or more devices failed to register. Details for each failure can be found
+     * in this exception. Any devices not found in the map of registration exceptions provided by this exception have registered successfully.
+     * @throws com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException If this operation takes longer than the provided timeout allows.
+     * @throws MultiplexingClientException If any other Exception is thrown, it will be nested into this exception.
      * @param deviceClients The device clients to associate with this multiplexing client.
+     * @param timeoutMilliseconds How long (in milliseconds) to let this operation wait for all registrations to complete. If this threshold is passed, a {@link com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException} is thrown.
      */
-    public void registerDeviceClients(Iterable<DeviceClient> deviceClients) throws InterruptedException {
+    public void registerDeviceClients(Iterable<DeviceClient> deviceClients, long timeoutMilliseconds) throws InterruptedException, MultiplexingClientException
+    {
         Objects.requireNonNull(deviceClients);
 
+        if (timeoutMilliseconds <= 0)
+        {
+            throw new IllegalArgumentException("Cannot set a device registration timeout of less than 0 milliseconds");
```

Comment:
less or equal? can it be 0?

Reply:
If we allowed 0, it would immediately throw a timeout exception, which doesn't seem useful. Using HTTP libraries as a model, setting a timeout to 0 usually means "infinite timeout" but I don't see a good use case for that when you can just set a really long timeout already.

Reply:
I mean it's a technicality issue, the sentence mentions it cannot be less than 0, the user can argue "it's not, it's 0" ... so maybe 
`Cannot set a device registration timeout of 0 milliseconds or less`

Reply:
Oh, good catch, I misunderstood what you were getting at